### PR TITLE
Added Configuring Zowe via JCL topic

### PR DESCRIPTION
**Pull request by lparadkar-rocket**
Describe your pull request here: Added Configuring Zowe via JCL

List the file(s) included in this PR: configuring-zowe-via-jcl, zos-components-installation-checklist

After creating the PR, follow the instructions in the comments.

### CODE REVIEW

**Review by Joe-Winchester — APPROVED** [review]

File: docs/user-guide/configuring-zowe-via-jcl.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+ # Configuring Zowe via JCL
+
+One option to configure Zowe is by directly customizing JCLs. The Zowe Runtime Dataset `SZWESAMP` contains JCL samples that have templates referencing Zowe YAML parameters. These samples should not be submitted without modification. Samples that are submitted without modification will end unsuccessfully with a JCL ERROR status.
+
+Edit and submit the job `SZWESAMP(ZWEGENER)` to validate the contents of your `zowe.yaml` before resolving the `JCL templates` and placing the resulting JCL into a separate data set created during installation. The location is specified in `zowe.setup.dataset.jcllib`.
```
Comment (Martin-Zeithaml) [inline docs/user-guide/configuring-zowe-via-jcl.md:5]: `jcllib` is not created by `zwe install`. This dataset is created by `zwe init (generate)`:
_...into a separate data set created by `ZWEGENER` job._

File: docs/user-guide/configuring-zowe-via-jcl.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+|APF Authorize privileged content | <br />**Purpose:**<br /> The majority of Zowe is unprivileged code running in key8. Zowe relies on a single component called ZIS to own all of the privileged code logic. The load library for the ZIS component and its extension library must be set as APF authorized and run in key 4 to use ZIS and components that depend upon it. <br /> **Action:**<br />1) APF authorize the datasets defined at `zowe.setup.dataset.authLoadlib` and `zowe.setup.dataset.authPluginLib`. <br />2) Define PPT entries for the members `ZWESIS01` and `ZWESAUX` as Key 4, NOSWAP in the `SCHEDxx` member of the system PARMLIB. | [ZWEIAPF](https://github.com/zowe/zowe-install-packaging/tree/feature/v3/jcl/files/SZWESAMP/ZWEIAPF) <br />
```
Comment (Martin-Zeithaml) [inline docs/user-guide/configuring-zowe-via-jcl.md:12]: key8 -> Key 8

File: docs/user-guide/configuring-zowe-via-jcl.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ |(z/OS v2.4 ONLY) Create Zowe SAF Resource Class |  On z/OS v2.4, the SAF resource class for Zowe is not included, and must be created. This step is not needed on z/OS v2.5 and later versions. | RACF: [ZWEIRACZ](https://github.com/zowe/zowe-install-packaging/tree/feature/v3/jcl/files/SZWESAMP/ZWEIRACZ) <br />TSS: [ZWEITSSZ](https://github.com/zowe/zowe-install-packaging/tree/feature/v3/jcl/files/SZWESAMP/ZWEITSSZ) <br />ACF2: [ZWEIACFZ](https://github.com/zowe/zowe-install-packaging/tree/feature/v3/jcl/files/SZWESAMP/ZWEIACFZ)
+ Copy STC JCL to PROCLIB | <br />**Purpose:**<br /> The job ZWESLSTC runs Zowe's webservers. and The job ZWESISTC runs the APF authorized cross-memory server. The job ZWESASTC is started by ZWESISTC on an as-needed basis. <br /> **Action:**<br /> Copy the members ZWESLSTC, ZWESISTC, and ZWESASTC into your desired PROCLIB. If the job names are customized, also modify the YAML values of them in `zowe.setup.security.stcs`. | [ZWEISTC](ZWEISTC)
```
Comment (Martin-Zeithaml) [inline docs/user-guide/configuring-zowe-via-jcl.md:15]: The job ZWESLSTC runs Zowe's webservers. and The job ZWESISTC
->
The job ZWESLSTC runs Zowe's webservers and the job ZWESISTC

File: docs/user-guide/configuring-zowe-via-jcl.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+## Keyring Tasks
+**Certificate requirements**  
+Ensure that your Zowe keyring has the following elements:
+
+* **Private key & certificate pair**  
+The Zowe Servers will use this certificate. Ensure that the certificate either does not have the `Extended Key Usage` attribute, or alternatively, that the certificate does have `Extended Key Usage` with both `Server Authorization` and `Client Authorization` values.
```
Comment (Martin-Zeithaml) [inline docs/user-guide/configuring-zowe-via-jcl.md:22]: We might place a link to https://docs.zowe.org/stable/user-guide/configure-certificates/#extended-key-usage somewhere here.

File: docs/user-guide/configuring-zowe-via-jcl.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
+ # Configuring Zowe via JCL
+
+One option to configure Zowe is by directly customizing JCLs. The Zowe Runtime Dataset `SZWESAMP` contains JCL samples that have templates referencing Zowe YAML parameters. These samples cannot be submitted without modification. 
+Edit and submit the job `SZWESAMP(ZWEGENER)` to validate the contents of your Zowe `YAML` before resolving the `JCL templates` and placing the resulting JCL into a separate `PDSE` created during installation. The location is specified in `zowe.setup.dataset.jcllib`.
```
[the remark this answers is not in the document]
Comment (janan07) [reply docs/user-guide/configuring-zowe-via-jcl.md]: agree. done

[the remark this answers is not in the document]
Comment (janan07) [reply docs/user-guide/configuring-zowe-via-jcl.md]: changed as per your comment

File: docs/user-guide/configuring-zowe-via-jcl.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+|APF Authorize privileged content | <br />**Purpose:**<br /> Zowe contains one privileged component, ZIS, which enables the security model by which the majority of Zowe is unprivileged and in key 8. The load library for the ZIS component and its extension library must be set APF authorized and run in key 4 to use ZIS and components that depend upon it. <br /> **Action:**<br />1) APF authorize the datasets defined at zowe.setup.dataset.authLoadlib and zowe.setup.dataset.authPluginLib. <br />2) Define PPT entries for the members ZWESIS01 and ZWESAUX as Key 4, NOSWAP in the SCHEDxx member of the system PARMLIB. | [ZWEIAPF](https://github.com/zowe/zowe-install-packaging/tree/feature/v3/jcl/files/SZWESAMP/ZWEIAPF) <br />
```
[the remark this answers is not in the document]
Comment (janan07) [reply docs/user-guide/configuring-zowe-via-jcl.md]: agree. done

**Review by Martin-Zeithaml — APPROVED** [review]
Approved. This doc is referring only new samples, which were already added to 2.17. This can go to v2.

**Review by janan07 — APPROVED** [review]
Reviewed and edited by doc squad

File: .github/workflows/build-docs.yml [hunk 7]
```diff
@@ -6,12 +6,6 @@ on:
       - master
   pull_request:
 
-permissions:
-  deployments: write
-  issues: write
-  pull-requests: write
-  statuses: write
```
Comment (zFernand0) [inline .github/workflows/build-docs.yml:13]: @lparadkar-rocket 
Please revert these changes
Also, please avoid changing workflows without discussing with other squads 😋

**Review by zFernand0 — CHANGES_REQUESTED** [review]

File: .github/workflows/build-docs.yml (hunk 7 above)
Comment (janan07) [reply .github/workflows/build-docs.yml:13]: Agreed, but I don't see in the list of commits where these permissions were removed in the first place.

File: docs/user-guide/configuring-zowe-via-jcl.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+| Task | Description | Sample JCL|
+|------|-------------|-----------|
+|Create VSAM Dataset for Caching Service | **Action**: Create a RLS or NONRLS dataset for the caching service, and set the name into the YAML value `components.caching-service.storage.vsam.name` | [ZWECSVSM](https://github.com/zowe/zowe-install-packaging/tree/feature/v3/jcl/files/SZWESAMP/ZWECSVSM)
+
+JCL samples for removing Zowe configuration also exist.
+|Action | Sample JCL |
+|------|-----------|
+|Remove Instance Datasets | [ZWERMVS](https://github.com/zowe/zowe-install-packaging/tree/feature/v3/jcl/files/SZWESAMP/ZWERMVS)
+|Remove SAF Permissions | [ZWENOSEC](https://github.com/zowe/zowe-install-packaging/tree/feature/v3/jcl/files/SZWESAMP/ZWENOSEC)
+|Remove Keyring | [ZWENOKR](https://github.com/zowe/zowe-install-packaging/tree/feature/v3/jcl/files/SZWESAMP/ZWENOKR) |
```
Comment (Martin-Zeithaml) [inline docs/user-guide/configuring-zowe-via-jcl.md:69]: This is ESM (External Security Manager) dependent, there is no ZWENOKR anymore, but ZWENOKRA, ZWENOKRR and ZWENOKRT (last letter is ESM: A=ACF2, R=RACF and T=TSS).

File: docs/user-guide/configuring-zowe-via-jcl.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+ Grant SAF premissions | <br />**Purpose:**<br /> The STC accounts for Zowe need permissions for operating servers, and users need permissions for interacting with the servers. <br />**Action:**<br /> [Set SAF permissions for accounts](https://docs.zowe.org/stable/user-guide/assign-security-permissions-to-users#security-permissions-reference-table) | RACF: [ZWEIRAC](https://github.com/zowe/zowe-install-packaging/tree/feature/v3/jcl/files/SZWESAMP/ZWEIRAC) <br /> TSS: [ZWEITSS](https://github.com/zowe/zowe-install-packaging/tree/feature/v3/jcl/files/SZWESAMP/ZWEITSS) <br /> ACF2: [ZWEIACF](https://github.com/zowe/zowe-install-packaging/tree/feature/v3/jcl/files/SZWESAMP/SZWIACF) <br />
```
Comment (Martin-Zeithaml) [inline docs/user-guide/configuring-zowe-via-jcl.md:20]: SZWIACF -> ZWEIACF

File: docs/user-guide/configuring-zowe-via-jcl.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+ |(z/OS v2.4 ONLY) Create Zowe SAF Resource Class |  On z/OS v2.4, the SAF resource class for Zowe is not included, and must be created. This step is not needed on z/OS v2.5 and later versions. | RACF: [ZWEIRACZ](https://github.com/zowe/zowe-install-packaging/tree/feature/v3/jcl/files/SZWESAMP/ZWEIRACZ) <br />TSS: [ZWEITSSZ](https://github.com/zowe/zowe-install-packaging/tree/feature/v3/jcl/files/SZWESAMP/ZWEITSSZ) <br />ACF2: [ZWEIACFZ](https://github.com/zowe/zowe-install-packaging/tree/feature/v3/jcl/files/SZWESAMP/ZWEIACFZ)
+ Copy STC JCL to PROCLIB | <br />**Purpose:**<br /> The job ZWESLSTC runs Zowe's webservers. The job ZWESISTC runs the APF authorized cross-memory server. The job ZWESASTC is started by ZWESISTC on an as-needed basis. <br /> **Action:**<br /> Copy the members ZWESLSTC, ZWESISTC, and ZWESASTC into your desired PROCLIB. If the job names are customized, also modify the YAML values of them in `zowe.setup.security.stcs`. | [ZWEISTC](ZWEISTC)
```
Comment (Martin-Zeithaml) [inline docs/user-guide/configuring-zowe-via-jcl.md:22]: The correct link is 
`[ZWEISTC](https://github.com/zowe/zowe-install-packaging/blob/v2.x/staging/files/SZWESAMP/ZWEISTC)`

File: .github/workflows/build-docs.yml (hunk 7 above)
Comment (zFernand0) [reply .github/workflows/build-docs.yml]: @lparadkar-rocket, I'm sorry for incorrectly tagging you.

After looking through the history a bit more closely, I noticed that it was someone else.
My apologies 😢
